README.md
[![Build Status](https://img.shields.io/travis/eximius313/bir1-api.svg)](https://travis-ci.org/eximius313/bir1-api) [![Bintray](https://api.bintray.com/packages/eximius313/bir1-api/maven/images/download.svg)](https://bintray.com/eximius313/bir1-api/maven)
    
# Baza Internetowa REGON - Java API

Java API do [Bazy Internetowej REGON](https://wyszukiwarkaregon.stat.gov.pl/appBIR/index.aspx)

## Podstawowe informacje:
http://bip.stat.gov.pl/dzialalnosc-statystyki-publicznej/rejestr-regon/interfejsyapi/

## Instrukcja techniczna:
[PDF](https://cdn.rawgit.com/eximius313/bir1-api/d841d6a4/docs/regon%20-%20instrukcja%20techniczna%20BIR1%20dla%20podm.komercyjnych_010.pdf)

## DEMO:
<details open>
<summary>Java</summary>

```java
public class Main {

    private static final String GUS_NIP = "5261040828";
    private static final String INVALID_NIP = "1261040828";

    public static void main(String[] args) throws IOException {
        try (final Connection connection = BIR1.createNewTestConnection()) {
            final ValueHandler valueHandler = connection.getValue();
            System.out.println("Service status: " + valueHandler.getServiceStatus());
            System.out.println("Session status: " + valueHandler.getSessionStatus());

            final DataHandler dataHandler = connection.findData();
            final Company company = dataHandler.getDataByNip(GUS_NIP).orElse(null);
            System.out.println("Found company: " + company.getName());

            final Company invalid = dataHandler.getDataByNip(INVALID_NIP).orElse(null);
            System.out.println("Invalid: " + invalid);
            System.out.println("Last message code: " + valueHandler.getMessageCode());
            System.out.println("Last message: " + valueHandler.getMessageContent());

            final ReportHandler reportHandler = connection.getReport();
            String fullReport = reportHandler.getFullReport(company.getRegon(), ReportType.PublDaneRaportPrawna)
                    .orElse(null);
            System.out.println("Full report:\n" + fullReport);
        }
    }
}
```
</details>


docs/release-notes.md
<sup><sup>*Release notes were automatically generated by [Shipkit](http://shipkit.org/)*</sup></sup>

#### 0.0.9
 - 2020-01-30 - [1 commit](https://github.com/eximius313/bir1-api/compare/v0.0.8...v0.0.9) by [Kamil](https://github.com/eximius313) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.0.9-green.svg)](https://bintray.com/eximius313/bir1-api/maven/0.0.9)
 - No pull requests referenced in commit messages.

#### 0.0.8
 - 2020-01-30 - [1 commit](https://github.com/eximius313/bir1-api/compare/v0.0.7...v0.0.8) by [Kamil](https://github.com/eximius313) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.0.8-green.svg)](https://bintray.com/eximius313/bir1-api/maven/0.0.8)
 - No pull requests referenced in commit messages.

#### 0.0.7
 - 2019-08-25 - [1 commit](https://github.com/eximius313/bir1-api/compare/v0.0.6...v0.0.7) by [Kamil](https://github.com/eximius313) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.0.7-green.svg)](https://bintray.com/eximius313/bir1-api/maven/0.0.7)
 - No pull requests referenced in commit messages.

#### 0.0.6
 - 2019-08-25 - [2 commits](https://github.com/eximius313/bir1-api/compare/v0.0.5...v0.0.6) by [Kamil](https://github.com/eximius313) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.0.6-green.svg)](https://bintray.com/eximius313/bir1-api/maven/0.0.6)
 - No pull requests referenced in commit messages.

#### 0.0.5
 - 2018-06-10 - [1 commit](https://github.com/eximius313/bir1-api/compare/v0.0.4...v0.0.5) by [Kamil](https://github.com/eximius313) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.0.5-green.svg)](https://bintray.com/eximius313/bir1-api/maven/0.0.5)
 - No pull requests referenced in commit messages.

#### 0.0.4
 - 2018-06-02 - [1 commit](https://github.com/eximius313/bir1-api/compare/v0.0.3...v0.0.4) by [Kamil](https://github.com/eximius313) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.0.4-green.svg)](https://bintray.com/eximius313/bir1-api/maven/0.0.4)
 - No pull requests referenced in commit messages.

#### 0.0.3
 - 2018-03-29 - [1 commit](https://github.com/eximius313/bir1-api/compare/v0.0.2...v0.0.3) by [Kamil](https://github.com/eximius313) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.0.3-green.svg)](https://bintray.com/eximius313/bir1-api/maven/0.0.3)
 - No pull requests referenced in commit messages.

#### 0.0.2
 - 2018-03-29 - [1 commit](https://github.com/eximius313/bir1-api/compare/v0.0.1...v0.0.2) by [Kamil](https://github.com/eximius313) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.0.2-green.svg)](https://bintray.com/eximius313/bir1-api/maven/0.0.2)
 - No pull requests referenced in commit messages.

#### 0.0.1
 - 2018-03-29 - 5 commits by [Kamil](https://github.com/eximius313) (4), [Marcin Stachniuk](https://github.com/mstachniuk) (1) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.0.1-green.svg)](https://bintray.com/eximius313/bir1-api/maven/0.0.1)
 - Add Shipkit [(#1)](https://github.com/eximius313/bir1-api/pull/1)



example/src/main/java/demo/Main.java
package demo;

import java.io.IOException;

import eximius.bir1.BIR1;
import eximius.bir1.Connection;
import eximius.bir1.data.Company;
import eximius.bir1.data.DataHandler;
import eximius.bir1.report.ReportHandler;
import eximius.bir1.report.ReportType;
import eximius.bir1.value.ValueHandler;

class Main {

    private static final String GUS_NIP = "5261040828";
    private static final String INVALID_NIP = "1261040828";

    public static void main(String[] args) throws IOException {
        try (final Connection connection = BIR1.createNewTestConnection()) {
            final ValueHandler valueHandler = connection.getValue();
            System.out.println("Service status: " + connection.getValue().getServiceStatus());
            System.out.println("Session status: " + valueHandler.getSessionStatus());

            final DataHandler dataHandler = connection.findData();
            final Company company = dataHandler.getDataByNip(GUS_NIP).orElse(null);
            System.out.println("Found company: " + company.getName());

            final Company invalid = dataHandler.getDataByNip(INVALID_NIP).orElse(null);
            System.out.println("Invalid: " + invalid);
            System.out.println("Last message code: " + valueHandler.getMessageCode());
            System.out.println("Last message: " + valueHandler.getMessageContent());

            final ReportHandler reportHandler = connection.getReport();
            String fullReport = reportHandler.getFullReport(company.getRegon(), ReportType.PublDaneRaportPrawna).orElse(null);
            System.out.println("Full report:\n" + fullReport);
        }
    }
}